pkg/gateway/filesystemwallet.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package gateway
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const dataFileExtension string = ".id"
+const extensionLength = 3
+
+// FileSystemWalletStore stores identity information used to connect to a Hyperledger Fabric network.
+// Instances are created using NewFileSystemWallet()
+type FileSystemWalletStore struct {
+	path string
+}
+
+// NewFileSystemWallet creates an instance of a wallet, held in memory.
+// This implementation is not backed by a persistent store.
+func NewFileSystemWallet(path string) (Wallet, error) {
```

Comment:
Why return interface instead of struct pointer from a constructor?

Reply:
fixed